src/main/java/seedu/address/model/person/Person.java

```diff
@@ -33,24 +35,28 @@ public class Person {
     /**
      * Every field must be present and not null.
      */
-    public Person(Name name, Phone phone, Email email, Set<Tag> tags, Appointment appointment, Property property) {
+    public Person(Name name, Phone phone, Email email, Set<Tag> tags, Role role, Appointment appointment, Property property) {
+        this.role = role;
         requireAllNonNull(name, phone, appointment, property);
```

Comment:
Ok, we should require role to be non null as well

Reply:
Rather than adding the role to the Person class, we could have added the role to only the Buyer and Seller class, that way you would have less code to rewrite as we take advantage of polymorphism. 
i.e We use the Person class to refer to either the Buyer or Seller class rather than explicitly handling the cases for a Buyer instance and a Seller instance